xml_converter/src/xml_converter.cpp

```diff
@@ -171,7 +192,15 @@ void process_data(
             }
         }
     }
-    if (duplicate_categories.size() > 0 && allow_duplicates == false) {
+
+    bool all_allow_duplicates_false = all_of(path_configs.begin(), path_configs.end(), [](const PathConfig p) {
+        if (p.type == "export") {
+            return p.allow_duplicates == false;
+        }
+        return true;
+    });
```

Comment:
This is a very convoluted block of code:
"We will not ignore duplicates if and only if the allow duplicates flag was not present after an export format"

When it probably should be:
"We will ignore duplicates if the allow duplicates flag was passed in"

The root cause is in the parser logic, only supporting flags by assigning them to `PathConfig`s.

The current implementation also will intentionally silently consume and ignore the allow duplicates flag if it was passed in to an input file. I dont think the user interactions were considered here, please break out the different scenarios from a user perspective and what the user would expect.

Reply:
I changed the design for the output of the argument parser to a class that contains a vector of PathConfigs and the global variables and made `--allow-duplicates` a global. This way it just needs to check for the one value.